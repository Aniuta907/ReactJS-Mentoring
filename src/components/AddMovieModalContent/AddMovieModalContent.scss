.add-movie {
  &-label {
    color: #f65161;
    font-size: large;
    font-family: "Open Sans", sans-serif;
    text-transform: uppercase;
    margin: 15px 0;
    letter-spacing: 1px;
  }

  &-input {
    background: #2e2e2e;
    color: #ffffffc9;
    font-size: x-large;
    font-family: "Open Sans", sans-serif;
    height: 50px;
    border: none;
    padding: 0 15px;
    letter-spacing: 1px;
    margin-bottom: 15px;
  }

  &-select {
    background: #2e2e2e;
    font-size: x-large;
    font-family: "Open Sans", sans-serif;
    height: 50px;
    border: none;
    padding: 0 15px;
    letter-spacing: 1px;
    margin-bottom: 15px;
    color: #ffffff93;
  }

  &-option {
    padding: 0 15px;
    outline: transparent;
    border: none;
    color: #ffffffc9;
  }
}

input[type="date"] {
  color: #555555;
}
