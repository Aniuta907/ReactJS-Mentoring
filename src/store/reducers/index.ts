import { combineReducers } from "redux";
import { TypedUseSelectorHook, useSelector } from 'react-redux'

import { moviesReducer } from "./movies";

export const rootReducer = combineReducers({
  moviesData: moviesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
