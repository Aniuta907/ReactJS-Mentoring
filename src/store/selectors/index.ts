export { selectMovies } from "./movies";
export { selectEditMovie } from "./editMovie";
export { selectDeleteMovie } from "./deleteMovie";
export { selectAddMovie } from "./addMovie";
