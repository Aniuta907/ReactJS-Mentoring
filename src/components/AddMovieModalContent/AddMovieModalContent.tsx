import React from "react";

import "./AddMovieModalContent.scss";
import { MovieGenreSelect } from "../MovieGenreSelect";

export class AddMovieModalContent extends React.Component {
  render(): React.ReactNode {
    return (
      <>
        <label className="add-movie-label">title</label>
        <input className="add-movie-input" placeholder="Title here" />

        <label className="add-movie-label">release date</label>
        <input type="date" className="add-movie-input" />

        <label className="add-movie-label">movie url</label>
        <input className="add-movie-input" placeholder="Movie URL here" />

        <label className="add-movie-label">genre</label>

        <MovieGenreSelect />

        <label className="add-movie-label">overview</label>
        <input className="add-movie-input" placeholder="Overview here" />

        <label className="add-movie-label">runtime</label>
        <input className="add-movie-input" placeholder="Runtime here" />
      </>
    );
  }
}
