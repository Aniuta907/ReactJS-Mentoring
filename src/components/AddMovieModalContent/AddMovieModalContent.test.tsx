import * as React from "react";
import { Provider } from 'react-redux';
import { shallow, ShallowWrapper } from 'enzyme';
import * as reactRedux from 'react-redux';

import { AddMovieModalContent } from './AddMovieModalContent';
import { store } from "../../store";

const ReduxProvider = ({ children, store }) => (
    <Provider store={store}>{children}</Provider>
)

describe('AddMovieModalContent', () => {
    it('should render correctly', () => {
        const props = {
            closeModal: jest.fn(),
        };        
        const wrapper: ShallowWrapper = shallow(
        <ReduxProvider store={store}>
            <AddMovieModalContent {...props} />
        </ReduxProvider>);
        expect(wrapper).toMatchSnapshot();
    });

    it('onSubmit should work correctly', () => {
        const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
        const addMovie = jest.fn();
        useDispatchMock.mockReturnValue(addMovie);
        expect(addMovie).not.toHaveBeenCalled();
    });
});
