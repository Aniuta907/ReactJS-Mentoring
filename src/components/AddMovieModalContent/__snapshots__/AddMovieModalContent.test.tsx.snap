// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModalContent should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AddMovieModalContent
    closeModal={[MockFunction]}
  />
</Provider>
`;
